### PR TITLE
[Security] Update Security module based on .NET SDK track1, using source code

### PR DESCRIPTION
**Pull request by yibirnba**
## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by Nickcandy** [comment]
@yibirnba Hi, there's still some error in your example:
 "Az.Security","Microsoft.Azure.Commands.Security.Cmdlets.SecurityContacts.GetSecurityContacts","Get-AzSecurityContact","0","3000","The type of property 'AlertNotifications' of type 'Microsoft.Azure.Commands.Security.Models.SecurityContacts.PSSecurityContact' has changed from 'System.String' to 'Microsoft.Azure.Commands.SecurityCenter.Models.SecurityContact.PSSecurityContactPropertiesAlertNotifications'.","Change the type of property 'AlertNotifications' back to 'System.String'."
  "Az.Security","Microsoft.Azure.Commands.Security.Cmdlets.SecurityContacts.GetSecurityContacts","Get-AzSecurityContact","0","3010","The property 'AlertsToAdmins' of type 'Microsoft.Azure.Commands.Security.Models.SecurityContacts.PSSecurityContact' has been removed.","Add the property 'AlertsToAdmins' back to type 'Microsoft.Azure.Commands.Security.Models.SecurityContacts.PSSecurityContact'."
  "Az.Security","Microsoft.Azure.Commands.Security.Cmdlets.SecurityContacts.RemoveSecurityContacts","Remove-AzSecurityContact","0","3000","The type of property 'AlertNotifications' of type 'Microsoft.Azure.Commands.Security.Models.SecurityContacts.PSSecurityContact' has changed from 'System.String' to 'Microsoft.Azure.Commands.SecurityCenter.Models.SecurityContact.PSSecurityContactPropertiesAlertNotifications'.","Change the type of property 'AlertNotifications' back to 'System.String'."
  "Az.Security","Microsoft.Azure.Commands.Security.Cmdlets.SecurityContacts.RemoveSecurityContacts","Remove-AzSecurityContact","0","3010","The property 'AlertsToAdmins' of type 'Microsoft.Azure.Commands.Security.Models.SecurityContacts.PSSecurityContact' has been removed.","Add the property 'AlertsToAdmins' back to type 'Microsoft.Azure.Commands.Security.Models.SecurityContacts.PSSecurityContact'."
  "Az.Security","Microsoft.Azure.Commands.Security.Cmdlets.SecurityContacts.SetSecurityContacts","Set-AzSecurityContact","0","3000","The type of property 'AlertNotifications' of type 'Microsoft.Azure.Commands.Security.Models.SecurityContacts.PSSecurityContact' has changed from 'System.String' to 'Microsoft.Azure.Commands.SecurityCenter.Models.SecurityContact.PSSecurityContactPropertiesAlertNotifications'.","Change the type of property 'AlertNotifications' back to 'System.String'."
  "Az.Security","Microsoft.Azure.Commands.Security.Cmdlets.SecurityContacts.SetSecurityContacts","Set-AzSecurityContact","0","3010","The property 'AlertsToAdmins' of type 'Microsoft.Azure.Commands.Security.Models.SecurityContacts.PSSecurityContact' has been removed.","Add the property 'AlertsToAdmins' back to type 'Microsoft.Azure.Commands.Security.Models.SecurityContacts.PSSecurityContact'."
 
Please correct them ASAP if you want to publish these code in this release. Code freeze would be at 28/11/2022

File: src/Security/Security/Models/SecurityContact/PSSecurityContact.cs [hunk 1]
```diff
@@ -24,8 +26,8 @@ public class PSSecurityContact
 
         public string Phone { get; set; }
 
-        public string AlertNotifications { get; set; }
+        public PSSecurityContactPropertiesAlertNotifications AlertNotifications { get; set; }
 
-        public string AlertsToAdmins { get; set; }
+        public PSSecurityContactPropertiesNotificationsByRole NotificationsByRole { get; set; }
```
Comment (BethanyZhou) [inline src/Security/Security/Models/SecurityContact/PSSecurityContact.cs:31]: As @Nickcandy mentioned, changing parameter type is regarded as breaking change. It wouldn't be allowed in regualer release. Please revert them. To work around the issue, how about adding two new parameters for these objects?

**Comment by isra-fel** [comment]
Kindly ping @yibirnba

**Comment by isra-fel** [comment]
Closing the PR for lack of activity. Feel free to reopen to continue working on it. Thanks 😀

**Comment by yibirnba** [comment]
Hi @***@***.***>
Work is still in progress there, adding @***@***.***> who track it. We are waiting to the owning team to solve the breaking change there.
Tal, is there an update?

________________________________
מאת: Yeming Liu ***@***.***>
נשלח: יום שישי, 27 בינואר 2023, 16:48
אל: Azure/azure-powershell ***@***.***>
עותק: Yifat Birnbaum ***@***.***>; Assign ***@***.***>
נושא: Re: [Azure/azure-powershell] [Security] Update Security module based on .NET SDK track1, using source code (PR #20164)


Closing the PR for lack of activity. Feel free to reopen to continue working on it. Thanks 😀

**Comment by talhers** [comment]
PR is still in progress

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
Please add change log for your changes.

**Review by BethanyZhou — APPROVED** [review]
LGTM

**Pull request merged**